src/server/ac_rot.rs

```diff
@@ -0,0 +1,167 @@
+// Copyright (c) 2021 Western Digital
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+//! A `manticore` "server" for a AC-RoT.
+//!
+//! This module provides structures for serving responses to a PA-RoT making
+//! requests to an AC-RoT.
+
+use crate::crypto::rsa;
+use crate::hardware;
+use crate::mem::Arena;
+use crate::net;
+use crate::server::options::Options;
+use crate::server::response::Response;
+use crate::server::rot::Rot;
+use crate::server::Error;
+
+/// A AC-RoT, or "Active Component's Root of Trust", server.
+///
+/// This type implements the request -> response "business logic" of the
+/// PA-Rot <-> AC-RoT interaction.
+pub struct AcRot<'a, Identity, Reset, Rsa> {
+    rot: Rot<'a, Identity, Reset, Rsa>,
+}
+
+impl<'a, Identity, Reset, Rsa> AcRot<'a, Identity, Reset, Rsa>
+where
+    Identity: hardware::Identity,
+    Reset: hardware::Reset,
+    Rsa: rsa::Builder,
+{
+    /// Create a new `AcRot` with the given `Options`.
+    pub fn new(opts: Options<'a, Identity, Reset, Rsa>) -> Self {
+        Self {
+            rot: Rot::new(opts),
+        }
+    }
+}
+
+impl<'a, Identity, Reset, Rsa> Response for AcRot<'a, Identity, Reset, Rsa>
+where
+    Identity: hardware::Identity,
+    Reset: hardware::Reset,
+    Rsa: rsa::Builder,
+{
+    fn process_request<'req>(
+        &mut self,
+        host_port: &mut dyn net::HostPort,
```

Comment:
Any particular reason for not using `impl HostPort`? In general I've preferred to not force the use of a `dyn` unless I can't avoid it, and allow clients to switch to vtables if they need it for code size reasons. WDYT?

(Oh, apparently I wrote this code. Lol. Up to you whether to add yet another type param.)

Reply:
Yeah, this is just based on what is already there. A trait is probably a better idea but I think that should be a separate change.

Reply:
Ack